modules/programs/noti.nix

```diff
@@ -0,0 +1,55 @@
+{ config, lib, pkgs, ...}:
+
+with lib;
+
+let
+
+  cfg = config.programs.noti;
+
+in
+
+{
+  meta.maintainers = [ maintainers.marsam ];
+
+  options.programs.noti = {
+    enable = mkEnableOption "Noti";
+
+    settings = mkOption {
+      type = types.attrs;
```

Comment:
Had a quick look at [noti.md](https://github.com/variadico/noti/blob/master/docs/noti.md) and from what I can tell all options expect a string value so here you could use `types.attrsOf types.str`.